src/fpm/graph.py

```diff
@@ -257,6 +257,42 @@ def get_3d_structure(g, element="Wall", threshold=0.05):
     return elements
 
 
+def get_internal_walls(g):
+    coords_m = get_coordinates_map(g)
+    print("Getting internal walls...")
+    wall_planes_by_space = dict()
+    for s, r, w in g.triples((None, FP["walls"], None)):
+        wall_ptr = g.value(s, FP["walls"])
+        wall_nodes = get_list_from_ptr(g, wall_ptr)
```

Comment:
If you are not recursively parsing lists of lists, you can pass the list's head (should be a [`BNode`](https://rdflib.readthedocs.io/en/7.1.1/apidocs/rdflib.html#rdflib.BNode)) into [`rdflib.graph.Graph.items()`](https://rdflib.readthedocs.io/en/7.1.1/apidocs/rdflib.html#rdflib.graph.Graph.items) to get the list back. So the `get_list_from_ptr` is rather redundant, but this maybe a separate issue.

If you want to parse list of lists recursively, then there's [`rdf_utils.collection.load_list_re`](https://secorolab.github.io/rdf-utils/reference/rdf_utils/collection/#rdf_utils.collection.load_list_re) in `rdf_utils`.

Reply:
Yeah, that's a remnant of the old code base, I've had it in my eye to refactor it because those two lines are used everywhere. Opened #14 to track this, thanks!